contributor-docs/versioning.md

```diff
@@ -132,6 +134,67 @@ In this situation, changing the layout of the `button` to `flex` would collapse
 the whitespace present between the text "Save" and "changes" and would be
 considered a breaking change.
 
+### A component changes its usage of a CSS Custom Property
+
+semver bump: potentially **major**
+
+When a component changes how it uses a CSS Custom Property, the change could be
+considered breaking if the CSS Custom Property is a part of the component's
+public API or participates in a broader system for design tokens. The breakage
+occurs when a consumer of the component is setting the value of this CSS Custom
+Property in some way expecting a change in the component. For example:
+
+```diff
+/* ExampleComponent.css */
+.ExampleComponent {
+  display: flex;
+-  column-gap: var(--example-component-gap, 1rem);
++  column-gap: var(--gap, 1rem);
+}
+```
+
+```css
+/* Consumer.css */
+.ExampleClassName {
+  --example-component-gap: var(--gap-none);
+}
+```
+
+In this case, the consumer of the component is setting the value of `--example-component-gap` but this will no longer have an effect after the change. There is an expectation that `--example-component-gap` is a part of the public API and therefore this change is considered breaking.
+
+At a high level, CSS Custom Properties in components typically come from:
+
+- Primer Primitives as design tokens for the design system
+- Component-level tokens which can be used to style, customize the component or
+  is used internally by the component
+
+As a rule of thumb, changes to CSS Custom Properties that use values from Primer
+Primitives are **not** a breaking change unless the token only exists in a new
+major version of Primer. In which case, a fallback should be used.
+
+```diff
+// This is not a breaking change if base-size-6 is defined in the current major
+// version of Primer Primitives
+.ExampleComponent {
+-  margin-block: var(--space-small, 0.375rem);
++  margin-block: var(--base-size-6, 0.375rem);
+}
+```
+
+```diff
+// This is a breaking change if base-size-6 is defined in the next major
```

Comment:
So this is a breaking change even though `--space-small` and `--base-size-6` come from Primer Primtives _and_ we provided a fallback?

Reply:
That's a great point about the fallback. You're totally right that the fallback would still apply so there would not be any breakage in the UI, in this case what I'm trying to think through is if this would be a breaking change when someone anticipated `--space-small` to be the token applied to a component and then it changes to a different token.

Put another way:

- `ExampleComponent` depends on `space-small`
- Consumer uses/leverage/extends `space-small`
- `ExampleComponent` no longer depends on `space-small`, uses `base-size-6`

I'm down either way on this, totally happy to say that the relationship between Component -> token can change and should not be a part of the contract here.

Reply:
I think that the relationship between Component and token can chance and should not be part of the contract here.